app/models/user.rb

```diff
@@ -969,28 +969,24 @@ def crypt_password # :nodoc:
   def user_requirements # :nodoc:
     if login.to_s.blank?
       errors.add(:login, :validate_user_login_missing.t)
-    elsif login.length < 3 || login.bytesize > 40
+    elsif login.length < 3 || login.size > 40
       errors.add(:login, :validate_user_login_too_long.t)
     elsif (other = User.find_by_login(login)) && (other.id != id)
       errors.add(:login, :validate_user_login_taken.t)
     end
 
-    if password.to_s.blank?
-      # errors.add(:password, :validate_user_password_missing.t)
-    elsif password.length < 5 || password.bytesize > 40
+    if password.to_s.present? && (password.length < 5 || password.size > 40)
```

Comment:
At this point is `password` the real password or the hash?  
If it's the hash then can we validate its size here?  Also see same issue in Account Controller.